pkg/analytics/up.go

```diff
@@ -0,0 +1,150 @@
+		"isV2":                       u.isV2,
+		"manifestType":               u.manifestType,
+		"isOktetoRepository":         u.isOktetoRepository,
+		"hasDependenciesSection":     u.hasDependenciesSection,
+		"hasBuildSection":            u.hasBuildSection,
+		"hasDeploySection":           u.hasDeploySection,
+		"hasReverse":                 u.hasReverse,
+		"mode":                       u.mode,
+		"failActivate":               u.failActivate,
+		"activateDurationSeconds":    u.activateDuration.Seconds(),
+		"initialSyncDurationSeconds": u.initialSyncDuration.Seconds(),
+		"isReconnect":                u.isReconnect,
+		"reconnectCause":             u.reconnectCause,
+		"errSync":                    u.errSync,
+		"errResetDatabase":           u.errResetDatabase,
+	}
+}
+
+// ManifestProps adds the tracking properties of the repository manifest
+func (u *UpMetricsMetadata) ManifestProps(m *model.Manifest) {
+	u.isV2 = m.IsV2
+	u.manifestType = m.Type
+	u.hasDependenciesSection = m.HasDependenciesSection()
+	u.hasBuildSection = m.HasBuildSection()
+	u.hasDeploySection = m.HasDeploySection()
+}
+
+// DevProps adds the tracking properties of the service development manifest
+func (u *UpMetricsMetadata) DevProps(d *model.Dev) {
+	u.hasReverse = len(d.Reverse) > 0
+	u.mode = d.Mode
+	u.isInteractive = d.IsInteractive()
+}
+
+// RepositoryProps adds the tracking properties of the repository
+func (u *UpMetricsMetadata) RepositoryProps(isOktetoRepository bool) {
+	u.isOktetoRepository = isOktetoRepository
+}
+
+// FailActivate sets to true the property failActivate
+func (u *UpMetricsMetadata) FailActivate() {
+	u.failActivate = true
+}
+
+// ActivateDuration adds the duration of up activation
+func (u *UpMetricsMetadata) ActivateDuration(duration time.Duration) {
+	u.activateDuration = duration
+}
+
+// InitialSyncDuration adds the duration of the initial sync
+func (u *UpMetricsMetadata) InitialSyncDuration(duration time.Duration) {
+	u.initialSyncDuration = duration
+}
+
+const (
+	// reconnectCauseDefault is the default cause for a reconnection
+	reconnectCauseDefault = "unrecognised"
+
+	// reconnectCauseDevPodRecreated is cause when pods UID change between retrys
+	reconnectCauseDevPodRecreated = "dev-pod-recreated"
+)
+
+// ReconnectDefault sets to true the property isReconnect and adds the cause "unrecognised"
+func (u *UpMetricsMetadata) ReconnectDefault() {
+	u.isReconnect = true
+	u.reconnectCause = reconnectCauseDefault
+}
+
+// ReconnectDevPodRecreated sets to true the property isReconnect and adds the cause "dev-pod-recreated"
+func (u *UpMetricsMetadata) ReconnectDevPodRecreated() {
+	u.isReconnect = true
+	u.reconnectCause = reconnectCauseDevPodRecreated
+}
+
+// ErrSync sets to true the property errSync
+func (u *UpMetricsMetadata) ErrSync() {
+	u.errSync = true
+}
+
+// ErrResetDatabase sets to true the property errResetDatabase
+func (u *UpMetricsMetadata) ErrResetDatabase() {
+	u.errResetDatabase = true
+}
```

Comment:
```suggestion
func (u *UpMetricsMetadata) ErrResetSyncDatabase() {
	u.errResetSyncDatabase = true
}
```
WDYT about add more context to the func, vars and metric id about the "reset database" event related to Syncthing?

Reply:
yes i did that at the followup PR https://github.com/okteto/okteto/pull/3876/files#diff-2a9738d7a31893670e28d1ce2d082069a5af8a0f5de048a12b02e175bee04becR155 🙏🏻